api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/AbstractAccountServiceImpl.java

```diff
@@ -88,55 +97,95 @@ public AccountList getAllAccounts(String columnName, int page, int size, String
 
     @Override
     public AccountList getAllAccountsByFilter(PluginRequestBody reqBody) {
-        String search = StringUtils.isEmpty(reqBody.getSearchtext()) ? "" : reqBody.getSearchtext();
         Map<String, String> sortOrderFilter = reqBody.getSortFilter();
         String sortElement = AdminConstants.ACCOUNT_ID;
         String sortOrder = AdminConstants.ASC;
-        if(sortOrderFilter != null &&  sortOrderFilter.containsKey(AdminConstants.SORT_ELEMENT)){
-            sortElement=  sortOrderFilter.get(AdminConstants.SORT_ELEMENT);
-            sortOrder = sortOrderFilter.containsKey(AdminConstants.SORT_ORDER) ? sortOrderFilter.get(AdminConstants.SORT_ORDER) :  AdminConstants.ASC;
+        if (sortOrderFilter != null && sortOrderFilter.containsKey(AdminConstants.SORT_ELEMENT)) {
+            sortElement = sortOrderFilter.get(AdminConstants.SORT_ELEMENT).equalsIgnoreCase("assets") ||
+                    sortOrderFilter.get(AdminConstants.SORT_ELEMENT).equalsIgnoreCase("violations")
+                    ? "cast("+sortOrderFilter.get(AdminConstants.SORT_ELEMENT)+" as decimal(10,2))" : sortOrderFilter.get(AdminConstants.SORT_ELEMENT);
+            sortOrder = sortOrderFilter.containsKey(AdminConstants.SORT_ORDER) ? sortOrderFilter.get(AdminConstants.SORT_ORDER) : AdminConstants.ASC;
         }
-        if(reqBody.getFilter() == null){
-            return convertToMap(accountsRepository.findAll(PageRequest.of(reqBody.getPage(), reqBody.getSize(),
-                            Sort.by(sortOrder.equalsIgnoreCase(AdminConstants.ASC) ? Sort.Direction.ASC : Sort.Direction.DESC, sortElement)),
-                    search.toLowerCase()));
+        String accountQuery = fetchQueryForAccountsFilterApplied(reqBody, sortOrder, sortElement);
+        List<Map<String, Object>> responseList = rdsRepository.getDataFromPacman(accountQuery);
+        List<AccountDetails> accountDetailsList = new ArrayList<>();
+        responseList.stream().forEach(obj -> {
+            AccountDetails accountDetails = new AccountDetails(
+                    (String) (obj.get("accountId") == null ? "N/A" : obj.get("accountId")),
+                    (String) (obj.get("accountName") == null ? "N/A" : obj.get("accountName")),
+                    (String) (obj.get("assets") == null ? "0" : obj.get("assets")),
+                    (String) (obj.get("violations") == null ? "0" : obj.get("violations")),
+                    (String) (obj.get("accountStatus") == null ? "N/A" : obj.get("accountStatus")),
+                    (String) (obj.get("platform") == null ? "N/A" : obj.get("platform")),
+                    (String) (obj.get("createdBy") == null ? "N/A" : obj.get("createdBy")),
+                    (Timestamp) (obj.get("createdTime")));
```

Comment:
Such type of constructor will not be used in other places because such type of 'obj' may not be present in other methods.  It will be less useful. Instead lets create a generic constructor like the above in AccountDetails. Derive all necessary attributes and use constructor to create object.